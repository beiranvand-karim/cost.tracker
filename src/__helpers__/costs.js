 const costs = [
   {
      cost: 1000,
      comment: 'this is a comment',
      category: 'CLOTHING'
   },
   {
      cost: 1500,
      comment: 'this is a comment',
      category: 'RENT'
   },
   {
      cost: 2000,
      comment: 'this is a comment',
      category: 'FOOD'
   }
];

export default costs;
