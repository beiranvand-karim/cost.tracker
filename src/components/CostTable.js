import {Table} from 'reactstrap';
import React from 'react';

export default function CostTable({costs}) {
   return (
      <Table responsive>
         <thead>
         <tr>
            <th>#</th>
            <th>Cost</th>
            <th>Category</th>
            <th>Comment</th>
         </tr>
         </thead>
         <tbody>
         {
            costs.map((cost, index) => (
               <tr key={index}>
                  <th scope="row">{index}</th>
                  <td>{cost.cost}</td>
                  <td>{cost.category}</td>
                  <td>{cost.comment}</td>
               </tr>
            ))
         }
         </tbody>
      </Table>
   )
}
