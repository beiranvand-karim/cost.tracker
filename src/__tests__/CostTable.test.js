/* global expect, describe, it, beforeEach */
import React from "react";
import renderer from "react-test-renderer";
import {shallow} from "enzyme";
import CostTable from "../components/CostTable";
import costs from "../__helpers__/costs";

describe('CostTable', () => {

   it('should render without crashing', () => {
      shallow(<CostTable costs={costs}/>)
   });

   it('should matches the snapshot', () => {
      const tree = renderer.create(<CostTable costs={costs}/>).toJSON();
      expect(tree).toMatchSnapshot()
   });

});
