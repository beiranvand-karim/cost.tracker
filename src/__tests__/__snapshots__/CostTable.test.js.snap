// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CostTable should matches the snapshot 1`] = `
<div
  className="table-responsive"
>
  <table
    className="table"
  >
    <thead>
      <tr>
        <th>
          #
        </th>
        <th>
          Cost
        </th>
        <th>
          Category
        </th>
        <th>
          Comment
        </th>
      </tr>
    </thead>
    <tbody>
      <tr>
        <th
          scope="row"
        >
          0
        </th>
        <td>
          1000
        </td>
        <td>
          CLOTHING
        </td>
        <td>
          this is a comment
        </td>
      </tr>
      <tr>
        <th
          scope="row"
        >
          1
        </th>
        <td>
          1500
        </td>
        <td>
          RENT
        </td>
        <td>
          this is a comment
        </td>
      </tr>
      <tr>
        <th
          scope="row"
        >
          2
        </th>
        <td>
          2000
        </td>
        <td>
          FOOD
        </td>
        <td>
          this is a comment
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;
